### PR TITLE
Add granite upgrade page to sidebar

### PR DESCRIPTION
**Pull request by pauldowman**
Adds granite upgrade page to the "Protocol Upgrades" sidebar section. Also removes footnote about granite not being approved yet.

### CODE REVIEW

File: specs/protocol/superchain-upgrades.md [hunk 1]
```diff
@@ -256,7 +256,7 @@ but the matching L1-origin information may not be present at the time of activat
   [Governance Proposal](https://gov.optimism.io/t/upgrade-proposal-5-ecotone-network-upgrade/7669).
 - `v7.0.0`: 2024 Jul 10th - Fjord network upgrade (RIP-7212 precompile + FastLZ cost fn + Brotli compression).
   [Governance Proposal](https://gov.optimism.io/t/upgrade-proposal-9-fjord-network-upgrade/8236).
-- `v8.0.0`: 2024 Aug 19th (optimistic[^1]) - Granite network upgrade (Limit ecpairing input size + Reduce Channel Timeout).
+- `v8.0.0`: 2024 Aug 19th - Granite network upgrade (Limit ecpairing input size + Reduce Channel Timeout).
```
Comment (ajsutton) [inline specs/protocol/superchain-upgrades.md]: I'm unclear why this is Aug 19th....

**Review by ajsutton — APPROVED** [review]
LGTM. The date confuses me a little but that hasn't changed in this PR so not a blocker.

File: specs/protocol/superchain-upgrades.md (hunk 1 above)
Comment (pauldowman) [reply specs/protocol/superchain-upgrades.md]: You're right. It was [August 16 for Sepolia](https://sepolia.etherscan.io/address/0xfd7E6Ef1f6c9e4cC34F54065Bf8496cE41A4e2e8) and will be today for mainnet. I'll fix.

**Comment by pauldowman** [comment]
Rebased onto main

File: specs/protocol/superchain-upgrades.md (hunk 1 above)
Comment (pauldowman) [reply specs/protocol/superchain-upgrades.md]: I mean tomorrow. Fixed.

**Review by Inphi — APPROVED** [review]

**Review by tynes — APPROVED** [review]

**Pull request merged**